### ISSUE TITLE
AssertionError

### ISSUE THREAD

**Issue by CarlTyler** [w2c/letsencrypt-esxi#5]
I run the command:  /etc/init.d/w2c-letsencrypt start

and I receive the following:
`Running 'start' action
Starting certificate renewal.
Existing cert issued for redirect.ovh.net but current domain name is ip-158-69-26.net. Requesting a new one!
Generating RSA private key, 4096 bit long modulus
...++++
....................................................................++++
e is 65537 (0x10001)
Serving HTTP on 0.0.0.0 port 8120 ...
Parsing account key...
Parsing CSR...
Found domains: ip-158-69-26.net
Getting directory...
Directory found!
Registering account...
Already registered! Account ID: https://acme-v02.api.letsencrypt.org/acme/acct/817945307
Creating new order...
Order created!
Verifying ip-158-69-26.net...
Traceback (most recent call last):
  File "./acme_tiny.py", line 145, in get_crt
    assert (disable_check or _do_request(wellknown_url)[0] == keyauthorization)
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./acme_tiny.py", line 199, in <module>
    main(sys.argv[1:])
  File "./acme_tiny.py", line 195, in main
    signed_crt = get_crt(args.account_key, args.csr, args.acme_dir, log=LOGGER, CA=args.ca, disable_check=args.disable_check, directory_url=args.directory_url, contact=args.contact, check_port=args.check_port)
  File "./acme_tiny.py", line 147, in get_crt
    raise ValueError("Wrote file to {0}, but couldn't download {1}: {2}".format(wellknown_path, wellknown_url, e))
ValueError: Wrote file to /opt/w2c-letsencrypt/.well-known/acme-challenge/SXykn4btnmIQbv-5OX-cS_fuIiMjUvuLlI0Yb92veKE, but couldn't download http://ip-158-69-26.net/.well-known/acme-challenge/SXykn4btnmIQbv-5OX-cS_fuIiMjUvuLlI0Yb92veKE:
Error: No cert obtained from Let's Encrypt. Generating a self-signed certificate.
hostd signalled.
rabbitmqproxy is not running
VMware HTTP reverse proxy signalled.
sfcbd-init: backgrounding ssl_reset on sfcbd-watchdog
vpxa signalled.
ssl_reset: vsanperfsvc is not running
/etc/init.d/vvold ssl_reset, PID 14994659
vvold is not running.
`

I believe the A record etc. is correct.  Any ideas? Thank you.

**Comment by Churro** [comment]
ip-158-69-26.net doesn't expose an ESXi installation and it seems the host is unreachable via HTTP(S).

**Comment by CarlTyler** [comment]
> ip-158-69-26.net doesn't expose an ESXi installation and it seems the host is unreachable via HTTP(S).

ip-158-69-26.net Is the domain name. Not the FQHN.   There are many boxes at the domain ip-158-69-26.net so it should not resolve to a specific machine, that’s what the fully qualified host name is for.

**Comment by Churro** [comment]
Exactly, and this is the reason it fails because a reachable FQDN is the prerequisite, see https://github.com/w2c/letsencrypt-esxi#prerequisites and

**Comment by CarlTyler** [comment]
> Exactly, and this is the reason it fails because a reachable FQDN is the prerequisite, see https://github.com/w2c/letsencrypt-esxi#prerequisites

I have a fully qualified hostname, and it is set in the ESXI configuration.  The FQHN resolves.

**Comment by CarlTyler** [comment]
> Exactly, and this is the reason it fails because a reachable FQDN is the prerequisite, see https://github.com/w2c/letsencrypt-esxi#prerequisites

Sorry for not sharing the fqhn here, I'm trying to avoid some bot finding it and banging on it.  I don't think there is a way to send a private message in github.

**Comment by Churro** [comment]
As can be seen in the initial log output, the script reads ip-158-69-26.net as the FQDN:
>  Existing cert issued for redirect.ovh.net but current domain name is ip-158-69-26.net. Requesting a new one!

Probably the fix is easy by just setting it again via `esxcfg-advcfg -s your-fancy-fqdn.com /Misc/hostname` and then calling `/etc/init.d/w2c-letsencrypt start`

**Comment by CarlTyler** [comment]
So that is how I had set the hostname, I should have set the FQHN, thank you, that did the trick.

In you example, you might want to put in a FQHN, example esxi.new-example.com  
https://github.com/w2c/letsencrypt-esxi#hostname-change

Thank you for your speedy response, this made things much easier.   FYI, in your instructions for the install, I did it via the WEB UI, I just pointed to your VIB url, as I wasn't sure how to reference a file in the datastore.  It now has the letsencrypt cert but Chrome is showing it as unsecure, but I can investigate that.  Thank you so much for your hard work.
